.github/workflows/preview-deployments.yml

```diff
@@ -0,0 +1,128 @@
+# CI to release the project for Linux, Windows, and MacOS
+
+name: Upload Preview Packages
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  macos:
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        python-version: ["3.7"]
```

Comment:
I am building on the oldest version instead of the latest version to ensure the most backward compatibility in every release